### PR TITLE
Windows has no direct way to kill a process based on port. 

### PR DESCRIPTION
**Pull request by TheBosZ**
Uses netstats and loops through the results to find the correct process to kill.

Also modify Process.run for the server to runInShell if on Windows.

Reference #36

### CODE REVIEW

File: lib/src/device.dart [hunk 1]
```diff
@@ -826,6 +826,26 @@ class AndroidDevice extends Device {
...
+
+      //List entries from netstat is formatted like so
+      // TCP    192.168.2.11:50945     192.30.252.90:443      LISTENING     1304
+      //This regexp is to find process where the the port exactly matches
+      RegExp pattern = new RegExp(':$_serverPort[ ]+');
+
+      //Split the columns by 1 or more spaces
+      RegExp columnPattern = new RegExp('[ ]+');
+      processes.forEach((String process){
+        if(process.contains(pattern)) {
```
Comment (abarth) [inline lib/src/device.dart]: Please add a space between "if" and "("

File: lib/src/device.dart [hunk 2]
```diff
@@ -826,6 +826,26 @@ class AndroidDevice extends Device {
...
+          //The last column is the Process ID
+          String taskNum = process.split(columnPattern).last;
```
Comment (abarth) [inline lib/src/device.dart]: s/taskNum/processId/

**Comment by abarth** [comment]
LGTM w/ style nits.

Please update PR with the changes all squashed together into a single commit.

**Comment by TheBosZ** [comment]
No problem, give me a second.

**Pull request closed without merging**